### PR TITLE
[BugFix] Enable AzKeyStore with keyring in Linux

### PR DESCRIPTION
**Pull request by msJinLei**
Bug fix for #20296 

* Enable AzKeystore with keyring in Linux
* Optimize the warning message for customers
* Update common library version.

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Review by BethanyZhou — APPROVED** [review]
LGTM

File: src/Accounts/Authentication/KeyStore/StorageWrapper.cs [hunk 1]
```diff
@@ -29,6 +30,13 @@ class StorageWrapper : IStorage
 
         private Storage _storage = null;
 
+        private bool isProtected;
```
Comment (isra-fel) [inline src/Accounts/Authentication/KeyStore/StorageWrapper.cs:33]: Let's name it _isProtected to be consistent with others.

File: src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs [hunk 2]
```diff
@@ -711,8 +713,8 @@ public void OnImport()
                 }
 
                 AzKeyStore keyStore = null;
-                //AzureSession.Instance.KeyStoreFile
-                keyStore = new AzKeyStore(AzureSession.Instance.ARMProfileDirectory, "keystore.cache", false, autoSaveEnabled);
+                keyStore = new AzKeyStore(AzureSession.Instance.ARMProfileDirectory, AzureSession.Instance.KeyStoreFile, false, autoSaveEnabled);
+                AzureSession.Instance.KeyStoreFile = keyStore.FileName;
```
[the remark this answers is not in the document]
Comment (msJinLei) [reply src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs:717]: FileName of keystore is finally assigned in the constructor of AzKeyStore according to different storage building properties.

[the remark this answers is not in the document]
Comment (msJinLei) [reply src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs]: Code removed

**Comment by msJinLei** [comment]
@microsoft-github-policy-service terminate

**Pull request merged**